### ISSUE TITLE
Possible Memory Leak

### ISSUE THREAD

**Issue by gonzabrusco** [emelianov/modbus-esp8266#208]
Hello. I think I've found a possible memory leak in your library.

https://github.com/emelianov/modbus-esp8266/blob/6600ea5187ab657be450e975fff29fbacefc28eb/src/ModbusTCPTemplate.h#L222

If the client is not connected, you do not FREE the new created object. 

My proposal:

```
CLIENT* currentClient = new CLIENT(c);
if (!currentClient) continue;
if(!currentClient->connected()) {
    delete currentClient;
    continue;
}
```

**Comment by emelianov** [comment]
Indeed it's very rare case (if even not impossible) but you are right thanks. Fixed in 2f14138.
Just keeping in mind that `!` is overridden it should be

```c
CLIENT* currentClient = new CLIENT(c);
if (!currentClient || !currentClient->connected()) {
	delete currentClient;
	continue;
}
```

**Comment by gonzabrusco** [comment]
Thanks!
